controllers/storagecluster/storageconsumer.go

```diff
@@ -98,26 +109,40 @@ func (s *storageConsumer) ensureCreated(r *StorageClusterReconciler, storageClus
 		return ctrl.Result{}, fmt.Errorf("failed to get available services configured in StorageCluster: %v", err)
 	}
 	consumerConfigMap := &corev1.ConfigMap{}
-	consumerConfigMap.Name = localStorageConsumerConfigMapName
+	consumerConfigMap.Name = storageConsumerObj.Spec.ResourceNameMappingConfigMap.Name
 	consumerConfigMap.Namespace = storageCluster.Namespace
 	if _, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, consumerConfigMap, func() error {
-		data := util.GetStorageConsumerDefaultResourceNames(
+		if consumerConfigMap.Data == nil {
+			consumerConfigMap.Data = map[string]string{}
+		}
+		resourceMap := util.WrapStorageConsumerResourceMap(consumerConfigMap.Data)
+		if availableServices.Rbd {
+			resourceMap.SetRbdRadosNamespaceName(util.ImplicitRbdRadosNamespaceName)
+			resourceMap.SetRbdClientProfileName("openshift-storage")
+		}
+		if availableServices.CephFs {
+			resourceMap.SetSubVolumeGroupName(subVolumeGroupName)
+			resourceMap.SetSubVolumeGroupRadosNamespaceName(subVolumeGroupName)
+			resourceMap.SetCephFsClientProfileName("openshift-storage")
+		}
+		if availableServices.Nfs {
+			resourceMap.SetNfsClientProfileName("openshift-storage")
+		}
+		defaultConsumerResourceNames := util.GetStorageConsumerDefaultResourceNames(
```

Comment:
is this correct? this overwrites the desired state of upgraded internal consumer, isn't it?